.github/workflows/build-images.yml

```diff
@@ -0,0 +1,77 @@
+name: build-images
+
+on:
+  push:
+    branches:
+      - edge
+  release:
+  workflow_dispatch:
+
+jobs:
+  build-container-image:
+    name: Image Build ${{ matrix.name }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - name: Debian
+            file: Docker/Dockerfile
+            flavor: |
+              latest=auto
+              suffix=
+              onlatest=false
+          - name: Alpine
+            file: Docker/Dockerfile-Alpine
+            flavor: |
+              latest=auto
+              suffix=-alpine
+              onlatest=true
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to DockerHub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    # - name: Login to GitHub Container Registry
+    #   uses: docker/login-action@v3
+    #   with:
+    #     registry: ghcr.io
+    #     username: ${{ github.repository_owner }}
+    #     password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Metadata
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        flavor: ${{ matrix.flavor }}
+        images: |
+          docker.io/freshrss/freshrss
+      #     ghcr.io/${{ github.repository }}
+        tags: |
+          type=edge
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}},enable=${{ !startsWith(github.ref, 'refs/tags/0.') }}
+        labels: |
+          org.opencontainers.image.description="A self-hosted RSS feed aggregator"
+          org.opencontainers.image.documentation="https://freshrss.github.io/FreshRSS/"
+          org.opencontainers.image.url="https://freshrss.org/"
```

Comment:
Personally I do the login as a last step before the build for two reasons.
The login can be time critical, if I remember correctly it only lasts for an hour for the ghcr.

Also I order my actions based on what they generate for the following actions. The fewer information the following actions have the better. A login is something that could be used otherwise so expose as few actions afterwards with it as possible. Probably irrelevant most of the time but I still try to follow that when creating workflows.

```suggestion
    - name: Metadata
      id: meta
      uses: docker/metadata-action@v5
      with:
        flavor: ${{ matrix.flavor }}
        images: |
          docker.io/freshrss/freshrss
      #     ghcr.io/${{ github.repository }}
        tags: |
          type=edge
          type=semver,pattern={{version}}
          type=semver,pattern={{major}}.{{minor}}
          type=semver,pattern={{major}},enable=${{ !startsWith(github.ref, 'refs/tags/0.') }}
        labels: |
          org.opencontainers.image.description="A self-hosted RSS feed aggregator"
          org.opencontainers.image.documentation="https://freshrss.github.io/FreshRSS/"
          org.opencontainers.image.url="https://freshrss.org/"

    - name: Login to DockerHub
      uses: docker/login-action@v3
      with:
        username: ${{ secrets.DOCKERHUB_USERNAME }}
        password: ${{ secrets.DOCKERHUB_TOKEN }}

    # - name: Login to GitHub Container Registry
    #   uses: docker/login-action@v3
    #   with:
    #     registry: ghcr.io
    #     username: ${{ github.repository_owner }}
    #     password: ${{ secrets.GITHUB_TOKEN }}
```